### PR TITLE
Improve PR info in default merge message

### LINKED ISSUE #9552: Merge commits generated by pull request capture no pull request detail
On Github, the merge commit generated from merging a pull request through the UI includes both the PR # and the description copy from the PR:

```
Merge pull request #{ISSUE_NUMBER} from {SOURCE_REPO}/{SOURCE_BRANCH}

{ISSUE_DESCRIPTION}
```

Gitea however generates a merge commit without any reference to the PR issue number:

```
Merge branch '{SOURCE_BRANCH}' of {SOURCE_REPO} into {TARGET_BRANCH}
```

Not that Gitea has to exactly replicate what Github does, but the inability to correlate merges back to their pull request easily creates a big pain point. Further, it is very helpful for the pull request description to make it into the history

### PR DESCRIPTION
**Pull request by shashvat-kedia**
Fix for issue #9552.

### CODE REVIEW

**Comment by 6543** [comment]
@sd1998 think you took the wrong issue NR

File: models/pull.go [hunk 1]
```diff
@@ -175,7 +175,11 @@ func (pr *PullRequest) GetDefaultMergeMessage() string {
 			return ""
 		}
 	}
-	return fmt.Sprintf("Merge branch '%s' of %s/%s into %s", pr.HeadBranch, pr.MustHeadUserName(), pr.HeadRepo.Name, pr.BaseBranch)
+	if err := pr.LoadIssue(); err != nil {
+		log.Error("Cannot load issue %d for PR id %d: Error: %v", pr.IssueID, pr.ID, err)
+		return ""
+	}
+	return fmt.Sprintf("Merge pull request %d: '%s' from '%s' of %s/%s into %s", pr.IssueID, pr.Issue.Title, pr.HeadBranch, pr.MustHeadUserName(), pr.HeadRepo.Name, pr.BaseBranch)
```
Comment (6543) [inline models/pull.go]:
```suggestion
	return fmt.Sprintf("Merge pull request '%s' (#%d) from %s/%s into %s", pr.Issue.Title, pr.Issue.Index, pr.MustHeadUserName(), pr.HeadBranch, pr.BaseBranch)
```
this is more like github + has still the PR title in it

Comment (zeripath) [reply models/pull.go]: You need index not ID

Comment (6543) [reply models/pull.go]: more :eyes: see more - added to suggestion

Comment (shashvat-kedia) [reply models/pull.go]: @6543 I have made the change.

**Review by 6543 — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Review by bagasme — APPROVED** [review]
OK, but can it be merged for v1.12?

**Comment by lunny** [comment]
@bagasme yes, it will.

**Comment by 6543** [comment]
can we change the PR **title** to a more specific meaning?

like "Make default merge message similar to github one" ...

**Comment by zeripath** [comment]
Make lg-tm work

**Comment by danwilliams** [comment]
@sd1998 @zeripath this PR does not fully satisfy ticket #9552, as it only changes the merge title, and does not put the PR description into the merge commit message body. Hence part of the issue remains outstanding.

However, my question is how to modify this without altering the source code and recompiling? I preferred the old format, before this change. Many people will prefer the new format. It seems to be possible to put templates into each repository, to override - but is there a way to apply a system-wide default? If so, this would allow the both my question and the remaining parts of the issue to be addressed.

Apologies if this is covered somewhere in documentation, but I could not find anything.

**Pull request merged**